mdm/api/src/main/java/com/gentics/mesh/core/data/node/HibNode.java

```diff
@@ -589,100 +188,183 @@ default String getPathSegment(String branchUuid, ContainerType type, String... l
 	 */
 	void setSchemaContainer(HibSchema container);
 
-	/**
-	 * Delete the node. Please use {@link #deleteFromBranch(InternalActionContext, HibBranch, BulkActionContext, boolean)} if you want to delete the node just from a specific branch.
-	 *
-	 * @param bac
-	 * @param ignoreChecks
-	 * @param recursive
-	 */
-	void delete(BulkActionContext bac, boolean ignoreChecks, boolean recursive);
-
-	/**
-	 * Handle the update tags request.
-	 *
-	 * @param ac
-	 * @param batch
-	 * @return Page which includes the new set of tags
-	 *
-	 */
-	// TODO Remove this method
-	Page<? extends HibTag> updateTags(InternalActionContext ac, EventQueueBatch batch);
-
-	/**
-	 * Update the tags of the node using the provides list of tag references.
-	 *
-	 * @param ac
-	 * @param batch
-	 * @param list
-	 * @return
-	 */
-	void updateTags(InternalActionContext ac, EventQueueBatch batch, List<TagReference> list);
-
-	/**
-	 * Return the breadcrumb nodes.
-	 *
-	 * @param ac
-	 * @return Deque with breadcrumb nodes
-	 */
-	Result<HibNode> getBreadcrumbNodes(InternalActionContext ac);
-
-	/**
-	 * Create the node specific delete event.
-	 *
-	 * @param uuid
-	 * @param schema
-	 * @param branchUuid
-	 * @param type
-	 * @param languageTag
-	 * @return Created event
-	 */
-	NodeMeshEventModel onDeleted(String uuid, HibSchema schema, String branchUuid, ContainerType type, String languageTag);
-
-	/**
-	 * Create a node tagged / untagged event.
-	 *
-	 * @param tag
-	 * @param branch
-	 * @param assignment
-	 *            Type of the assignment
-	 * @return
-	 */
-	NodeTaggedEventModel onTagged(HibTag tag, HibBranch branch, Assignment assignment);
-
 	/**
 	 * Check whether the node is the base node of its project
 	 *
 	 * @return true for base node
 	 */
 	boolean isBaseNode();
 
-	/**
-	 * Check whether the node is visible in the given branch (that means has at least one DRAFT graphfieldcontainer in the branch)
-	 *
-	 * @param branchUuid
-	 *            branch uuid
-	 * @return true if the node is visible in the branch
-	 */
-	boolean isVisibleInBranch(String branchUuid);
-
 	/**
 	 * Transform the node information to a minimal reference which does not include language or type information.
 	 *
 	 * @return
 	 */
-	NodeReference transformToMinimalReference();
+	default NodeReference transformToMinimalReference() {
+		NodeReference ref = new NodeReference();
+		ref.setUuid(getUuid());
+		ref.setSchema(getSchemaContainer().transformToReference());
+		return ref;
+	}
 
-	/**
-	 * Transform the node information to a version list response.
-	 *
-	 * @param ac
-	 * @return Versions response
-	 */
-	NodeVersionsResponse transformToVersionList(InternalActionContext ac);
+	default String getSubETag(InternalActionContext ac) {
```

Comment:
We can consider moving these default methods to the `PersistingNodeDao` and keep this interface clean

Reply:
Yes, we have to, since the implementation considers lower level code usage, and not just pure business logic.

Reply:
Though this requires some refactoring, especially in the GraphQL part, so I leave it as is for now.